test/src/middleware/navigation.middleware.test.ts

```diff
@@ -291,4 +291,168 @@ describe("navigiationMiddleware", () => {
         expect(response.redirect).toHaveBeenCalledWith("/authorised-agent/manage-users");
     });
 
+    it("should redirect to manage-users when referer is not in the allowed list for the edit member role", () => {
```

Comment:
I see you're copying what is already in this test file, but there's a good use case for parameterization here so we don't need to keep copy pasting entire tests and instead can easy just change parameters when adding new ones i.e

```ts
    describe("allowed navigation paths", () => {
        const allowedNavigationCases = [
            {
                description: "should allow admin user to check member details page when add a user is the referer",
                url: "/authorised-agent/check-member-details",
                referer: "/authorised-agent/add-user",
                user: adminUser
            },
            // ... Add more allowed navigation test cases here
        ];

        test.each(allowedNavigationCases)(
            "$description",
            ({ url, referer, user }) => {
                const mockedNext = jest.fn();
                const request = mockRequest();
                const response = mockResponse();

                request.originalUrl = url;
                request.headers.referer = referer;
                getExtraDataSpy.mockReturnValue(user);

                navigationMiddleware(request, response, mockedNext);

                expect(mockedNext).toHaveBeenCalled();
            }
        );
    });
```

Similar pattern can be used for redirect cases & for role-based access restrictions.

If this doesn't make sense though or is too much effort we could always do it later?